### ISSUE TITLE
RMSProp causing outputs to return NaN

### ISSUE THREAD

**Issue by EthanMacdonald** [dmlc/mxnet#5217]
I've noticed that RMSProp seems to cause my networks to return NaNs. This doesn't always happen on the first training batch, but the minimal example below starts returning NaNs right away.

In the example below, changing the kwargs of RMSProp (e.g., `clip_gradient=1.0`) doesn't seem to prevent the NaNs from occurring. If you uncomment line 5 and comment line 6, the script seems to run fine with SGD. I also tried it with Adam and that worked alright too.

## Minimal Example
```python
import numpy as np
import mxnet as mx

# Optimizer
# optimizer = mx.optimizer.SGD()
optimizer = mx.optimizer.RMSProp()

# Data
n_points = 10
data = np.linspace(0, 2, n_points)
label = data * 2
data = mx.io.NDArrayIter(data, label)

# Simple network
net = mx.sym.Variable("data")
net = mx.sym.FullyConnected(net, name="fc1", num_hidden=1)
net = mx.sym.LinearRegressionOutput(data=net, name='softmax')
net = mx.module.Module(net)
net.bind(data_shapes=data.provide_data, label_shapes=data.provide_label)
net.init_params()
net.init_optimizer(optimizer=optimizer)

# Train
for data_batch in data:
    net.forward_backward(data_batch)
    net.update()
print(net.predict(data).asnumpy())
```
```
[[ nan]
 [ nan]
 [ nan]
 [ nan]
 [ nan]
 [ nan]
 [ nan]
 [ nan]
 [ nan]
 [ nan]]
```

## Environment info
Operating System: OS X, Ubuntu 14.04

Package used (Python/R/Scala/Julia): Python

Python version and distribution: Python 3.5.2 :: Anaconda 4.2.0 (x86_64)

Git commit hash: b4e8743101294cf7e37510ebe4cdfd0238520156

**Comment by piiswrong** [comment]
@leezu @sxjscience

**Comment by leezu** [comment]
Hm, I cannot reproduce that behavior. Note that both the Python and C++ interface of RMSProp changed, so if you made a git pull recently and did not rebuilt the C++ side that might cause the behavior. If you did, can you try a `make clean` before building?

**Comment by sxjscience** [comment]
I also cannot reproduce the error.

**Comment by EthanMacdonald** [comment]
Thanks for the help! That helped me narrow things down. Last night I got it working on both operating systems. 

In case anyone has a similar problem in the future:
I had installed MXNet prior to RMSProp moving to NNVM. I used `git pull && update && make clean && make -j$(nproc)` to recompile the C++ interface after the new updates were introduced, and then updated the python interface. This is around the time that the NaNs started appearing, so I'm guessing there was a mismatch between the Python & C++ interface. Since it wasn't reproducible I decided to completely uninstall & reinstall everything from scratch (on both the Python and C++ side). I ran my scripts with RMSProp last night and they seem to be working fine.
